lib/nps-survey/classes/nps-survey-script.php

```diff
@@ -0,0 +1,509 @@
+	 *
+	 * @param  string $plugin_slug slug of unique NPS Survey.
+	 * @return array<string, mixed>
+	 */
+	public static function get_nps_survey_dismiss_status( string $plugin_slug ) {
+
+		$default_status = get_option(
+			self::get_nps_id( $plugin_slug ),
+			array(
+				'dismiss_count'       => 0,
+				'dismiss_permanently' => false,
+				'dismiss_step'        => '',
+				'dismiss_time'        => '',
+				'dismiss_timespan'    => null,
+				'first_render_time'   => null,
+			)
+		);
+
+		if ( ! is_array( $default_status ) ) {
+			return array();
+		}
+
+		return array(
+			'dismiss_count'       => ! empty( $default_status['dismiss_count'] ) ? $default_status['dismiss_count'] : 0,
+			'dismiss_permanently' => ! empty( $default_status['dismiss_permanently'] ) ? $default_status['dismiss_permanently'] : false,
+			'dismiss_step'        => ! empty( $default_status['dismiss_step'] ) ? $default_status['dismiss_step'] : '',
+			'dismiss_time'        => ! empty( $default_status['dismiss_time'] ) ? $default_status['dismiss_time'] : '',
+			'dismiss_timespan'    => ! empty( $default_status['dismiss_timespan'] ) ? $default_status['dismiss_timespan'] : null,
+			'first_render_time'   => ! empty( $default_status['first_render_time'] ) ? $default_status['first_render_time'] : null,
+		);
+	}
+
+	/**
+	 * Show status of NPS Survey.
+	 *
+	 * @param  string $plugin_slug slug of unique NPS Survey.
+	 * @param  int    $display_after number of days after which NPS Survey should be displayed.
+	 * @return bool
+	 */
+	public static function is_show_nps_survey_form( string $plugin_slug, int $display_after ) {
+
+		$current_time = time();
+		$status       = self::get_nps_survey_dismiss_status( $plugin_slug );
+
+		if ( $status['dismiss_permanently'] ) {
+			return false;
+		}
+
+		$first_render_time = $status['first_render_time'];
+
+		if ( 0 !== $display_after ) {
+			if ( null === $first_render_time ) {
+				$status['first_render_time'] = $current_time;
+				update_option( self::get_nps_id( $plugin_slug ), $status );
+				$status = self::get_nps_survey_dismiss_status( $plugin_slug );
+				return false;
+			}
+			if ( $display_after + $first_render_time > $current_time ) {
+				return false;
+			}
+		}
+
+		// Retrieve the stored date time stamp from wp_options.
+		$stored_date_timestamp = $status['dismiss_time'];
+		$dismiss_timespan      = $status['dismiss_timespan'];
+
+		if ( $stored_date_timestamp ) {
+
+			$current_time = time();
+
+			// time difference of current time and the time user dismissed the nps.
+			$time_difference = $current_time - $stored_date_timestamp;
+
+			// Check if two weeks have passed.
+			if ( $time_difference <= $dismiss_timespan ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get NPS Dismiss Option Name.
+	 *
+	 * @param string $plugin_slug Plugin name.
+	 * @return string
+	 */
+	public static function get_nps_id( $plugin_slug ) {
+		return 'nps-survey-' . $plugin_slug;
+	}
+   }
+ }
```

Comment:
Add this } at the end of the file, same for UABB Pro, I have merged the PR already